src/glyph/mod.rs

```diff
@@ -95,6 +100,113 @@ impl Glyph {
             lib: None,
         }
     }
+
+    /// Move libs from the lib's `public.objectLibs` into the actual objects.
+    /// The key will be removed from the glyph lib.
+    fn load_object_libs(&mut self) -> Result<(), ErrorKind> {
+        let mut object_libs =
+            match self.lib.as_mut().and_then(|lib| lib.remove(PUBLIC_OBJECT_LIBS_KEY)) {
+                Some(lib) => lib.into_dictionary().ok_or(ErrorKind::BadLib)?,
+                None => return Ok(()),
+            };
+
+        if let Some(anchors) = &mut self.anchors {
+            for anchor in anchors {
+                if let Some(lib) =
+                    anchor.identifier().and_then(|id| object_libs.remove(id.as_str()))
+                {
+                    let lib = lib.into_dictionary().ok_or(ErrorKind::BadLib)?;
+                    anchor.replace_lib(lib);
+                }
+            }
+        }
+
+        if let Some(guidelines) = &mut self.guidelines {
+            for guideline in guidelines {
+                if let Some(lib) =
+                    guideline.identifier().and_then(|id| object_libs.remove(id.as_str()))
+                {
+                    let lib = lib.into_dictionary().ok_or(ErrorKind::BadLib)?;
+                    guideline.replace_lib(lib);
+                }
+            }
+        }
+
+        if let Some(outline) = &mut self.outline {
+            for contour in &mut outline.contours {
+                if let Some(lib) =
+                    contour.identifier().and_then(|id| object_libs.remove(id.as_str()))
+                {
+                    let lib = lib.into_dictionary().ok_or(ErrorKind::BadLib)?;
+                    contour.replace_lib(lib);
+                }
+                for point in &mut contour.points {
+                    if let Some(lib) =
+                        point.identifier().and_then(|id| object_libs.remove(id.as_str()))
+                    {
+                        let lib = lib.into_dictionary().ok_or(ErrorKind::BadLib)?;
+                        point.replace_lib(lib);
+                    }
+                }
+            }
+            for component in &mut outline.components {
+                if let Some(lib) =
+                    component.identifier().and_then(|id| object_libs.remove(id.as_str()))
+                {
+                    let lib = lib.into_dictionary().ok_or(ErrorKind::BadLib)?;
+                    component.replace_lib(lib);
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Dump guideline libs into a Plist.
+    fn dump_object_libs(&self) -> Plist {
+        let mut object_libs = Plist::default();
+
+        let mut dump_lib = |id: Option<&Identifier>, lib: &Plist| {
+            let id = id.map(|id| id.as_str().to_string());
+            object_libs.insert(id.unwrap(), plist::Value::Dictionary(lib.clone()));
```

Comment:
Are there any possible circumstances where we don't have an identifier? What would they be?

Reply:
I don't know :O maybe if we decide to allow id removal at some point and mess the logic up.